include/WaveParallel.hpp

```diff
@@ -0,0 +1,338 @@
+#ifndef WAVE_EQUATION_PARALLEL_HPP
+#define WAVE_EQUATION_PARALLEL_HPP
+
+
+// ==================== INCLUDES ====================
+
+// Logging in parallel environment
+#include <deal.II/base/conditional_ostream.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/function.h>
+#include <deal.II/base/utilities.h>
+
+// Distributed Triangulation and Grid Generation
+#include <deal.II/distributed/fully_distributed_tria.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_in.h>
+
+// Finite Element space
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/fe_simplex_p.h>
+#include <deal.II/fe/fe_q.h>
+
+// Degrees of freedom
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+// Linear Algebra - Distributed
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/trilinos_precondition.h>
+#include <deal.II/lac/trilinos_sparse_matrix.h>
+
+// Numeric techniques
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/numerics/matrix_tools.h>
+
+// Timing
+#include <deal.II/base/timer.h>
+
+// STL libraries
+#include <iostream>
+#include <fstream>
+#include <memory>
+
+// ==================================================
+
+using namespace dealii;
+
+template <unsigned int dim>
+class WaveEquationParallel
```

Comment:
I am assuming that this class implements the Crank-Nicolson based solver, whereas the other one implements the solver based on Verlet integration.

If so, maybe the name of this class is not particularly explicative (also noting that there is no documentation comment that makes it obvious that this implements the Crank-Nicolson method).